File: app/src/main/java/com/yeyintlwin/musicsstore/MainController.java
package com.yeyintlwin.musicsstore;

import android.annotation.SuppressLint;
import android.app.Application;
import android.content.SharedPreferences;
import android.preference.PreferenceManager;

import com.aspsine.multithreaddownload.DownloadConfiguration;
import com.aspsine.multithreaddownload.DownloadManager;

public class MainController extends Application {
    private static SharedPreferences preferences;
    private static SharedPreferences.Editor editor;

    public static void putBoolean(String key, Boolean value) {
        editor.putBoolean(key, value).apply();
    }

    public static Boolean getBoolean(String key, Boolean defValue) {
        return preferences.getBoolean(key, defValue);
    }

    public static void putInt(String key, Integer value) {
        editor.putInt(key, value).apply();
    }

    public static Integer getInt(String key, Integer defValue) {
        return preferences.getInt(key, defValue);
    }

    public static void putString(String key, String value) {
        editor.putString(key, value).apply();
    }

    public static String getString(String key, String defValue) {
        return preferences.getString(key, defValue);
    }

    @SuppressLint("CommitPrefEdits")
    @Override
    public void onCreate() {
        super.onCreate();
        preferences = PreferenceManager.getDefaultSharedPreferences(this);
        editor = preferences.edit();
        initDownloader();
    }

    private void initDownloader() {
        DownloadConfiguration downloadConfiguration = new DownloadConfiguration();
        downloadConfiguration.setMaxThreadNum(10);
        downloadConfiguration.setThreadNum(1);
        DownloadManager.getInstance().init(this, downloadConfiguration);
    }
}

File: app/src/main/java/com/yeyintlwin/musicsstore/MainController.kt
package com.yeyintlwin.musicsstore

import android.annotation.SuppressLint
import android.app.Application
import android.content.SharedPreferences
import android.preference.PreferenceManager

import com.aspsine.multithreaddownload.DownloadConfiguration
import com.aspsine.multithreaddownload.DownloadManager

class MainController : Application() {

    @SuppressLint("CommitPrefEdits")
    override fun onCreate() {
        super.onCreate()
        preferences = PreferenceManager.getDefaultSharedPreferences(this)
        editor = preferences!!.edit()
        initDownloader()
    }

    private fun initDownloader() {
        val downloadConfiguration = DownloadConfiguration()
        downloadConfiguration.maxThreadNum = 10
        downloadConfiguration.threadNum = 1
        DownloadManager.getInstance().init(this, downloadConfiguration)
    }

    companion object {
        private var preferences: SharedPreferences? = null
        private var editor: SharedPreferences.Editor? = null

        fun putBoolean(key: String, value: Boolean?) {
            editor!!.putBoolean(key, value!!).apply()
        }

        fun getBoolean(key: String, defValue: Boolean?): Boolean {
            return preferences!!.getBoolean(key, defValue!!)
        }

        fun putInt(key: String, value: Int?) {
            editor!!.putInt(key, value!!).apply()
        }

        fun getInt(key: String, defValue: Int?): Int {
            return preferences!!.getInt(key, defValue!!)
        }

        fun putString(key: String, value: String) {
            editor!!.putString(key, value).apply()
        }

        fun getString(key: String, defValue: String): String? {
            return preferences!!.getString(key, defValue)
        }
    }
}
